Script/AtomicEditor/hostExtensions/HostExtensionServices.ts

```diff
@@ -206,21 +270,13 @@ export class UIServicesProvider extends ServicesProvider<Editor.HostExtensions.U
         super();
     }
 
-    private mainFrameMenu: MainFramMenu = null;
-    private hierarchyFrameMenu: HierarchyFrameMenu = null;
-    private projectFrameMenu: ProjectFrameMenu = null;
+    private mainFrame: MainFrame = null;
     private modalOps: ModalOps;
 
-    init(mainFrameMenu: MainFramMenu, hierarchyFrameMenu: HierarchyFrameMenu, projectFrameMenu: ProjectFrameMenu, modalOps: ModalOps) {
+    init(mainFrame: MainFrame, modalOps: ModalOps) {
```

Comment:
nice cleanup of the signature.  No need to pass all those elements in if they are all available on the main frame.